test/react-native-cli/features/fixtures/Dockerfile

```diff
@@ -1,15 +1,36 @@
 FROM node:lts-alpine
 
 RUN apk add git
+RUN npm i -g run-func
 
-COPY . /app
+# TODO(PLAT-5515) 'pod install' is currently always run but will only work on
+# macOS, so we stub it out with a script that does nothing
+RUN printf '#!/usr/bin/env sh\nprintf "this is a stub of $0\n"' > /usr/bin/pod && chmod +x /usr/bin/pod
 
 WORKDIR /app
 
-RUN npm i -g bugsnag-react-native-cli-*.tgz
+# Artifcatory authentication
+ARG REGISTRY_URL
+ARG REG_BASIC_CREDENTIAL
+ARG REG_NPM_EMAIL
+RUN npm config set registry $REGISTRY_URL
+RUN npm config set _auth $REG_BASIC_CREDENTIAL
+RUN npm config set email $REG_NPM_EMAIL
+RUN npm config set always-auth = true
```

Comment:
Artifactory hosts packages as `tgz` files, so I wonder whether we can keep the old way of installing the package, but provide a different tarball on CI? That way we can build the package locally without having to deal with artifactory at all — I think it's more valuable to be easier to run locally than on CI vs the other way around

e.g. currently to run all of these tests locally, all you need to do is run two commands & the only setup is to login to docker to get access the maze runner images (which will always be needed);

```sh
$ docker-compose build react-native-cli
$ REACT_NATIVE_VERSION=rn0_63 docker-compose run react-native-cli
```

If we do go this route then how to get this to work needs to be documented in `TESTING.md`

Reply:
Speaking to @bengourley a couple of times, the general feeling was that NPM publish and install from a registry, rather than using `pack` is more representative of real life.  It also means we can work towards a consistent approach across the repo (React Native cannot use `pack`).  It's also worth noting that you don't have to use Artifactory for that - Verdaccio is a lightweight NPM registry that serves nicely for local use.

That all said, we could consider providing both as options, assuming the plumbing didn't become too convoluted.  And you're certainly right about updating `TESTING.md` - it has been on my mind to do, but I'd been waiting for things to settle a bit and it probably becomes easier to do once the MazeRunner changes are released.

BTW, if you update the `Gemfile` locally to point at the MR integration branch (or local checkout), you don't need to build the docker image at all - you can just run:
```
REACT_NATIVE_VERSION=rn0_63 bundle exec maze-runner
```